federatedrec/param/factorization_machine_param.py

```diff
@@ -0,0 +1,289 @@
+        if not isinstance(self.tol, (int, float)):
+            raise ValueError(
+                "factorization_param's tol {} not supported, should be float type".format(self.tol))
+
+        if not isinstance(self.clip_gradient, (int, float, None)):
+            raise ValueError(
+                "factorization_param's clip_gradient {} not supported, should be float type".format(self.clip_gradient))
+
+        if type(self.alpha).__name__ not in ["float", 'int']:
+            raise ValueError(
+                "factorization_param's alpha {} not supported, should be float or int type".format(self.alpha))
+
+        if type(self.optimizer).__name__ != "str":
+            raise ValueError(
+                "factorization_param's optimizer {} not supported, should be str type".format(self.optimizer))
+        else:
+            self.optimizer = self.optimizer.lower()
+            if self.optimizer not in ['sgd', 'rmsprop', 'adam', 'adagrad', 'nesterov_momentum_sgd']:
+                raise ValueError(
+                    "factorization_param's optimizer not supported, optimizer should be"
+                    " 'sgd', 'rmsprop', 'adam', 'nesterov_momentum_sgd' or 'adagrad'")
+
+        if self.batch_size != -1:
+            if type(self.batch_size).__name__ not in ["int"] \
+                    or self.batch_size < consts.MIN_BATCH_SIZE:
+                raise ValueError(descr + " {} not supported, should be larger than 10 or "
+                                         "-1 represent for all data".format(self.batch_size))
+
+        if not isinstance(self.learning_rate, (float, int)):
+            raise ValueError(
+                "factorization_param's learning_rate {} not supported, should be float or  int type".format(
+                    self.learning_rate))
+
+        self.init_param.check()
+
+        if type(self.max_iter).__name__ != "int":
+            raise ValueError(
+                "factorization_param's max_iter {} not supported, should be int type".format(self.max_iter))
+        elif self.max_iter <= 0:
+            raise ValueError(
+                "factorization_param's max_iter must be greater or equal to 1")
+
+        if type(self.early_stop).__name__ != "str":
+            raise ValueError(
+                "factorization_param's early_stop {} not supported, should be str type".format(
+                    self.early_stop))
+        else:
+            self.early_stop = self.early_stop.lower()
+            if self.early_stop not in ['diff', 'abs', 'weight_diff']:
+                raise ValueError(
+                    "factorization_param's early_stop not supported, converge_func should be"
+                    " 'diff' or 'abs'")
+
+        self.encrypt_param.check()
+        if self.encrypt_param.method not in [consts.PAILLIER, None]:
+            raise ValueError(
+                "factorization_param's encrypted method support 'Paillier' or None only")
+
+        if type(self.decay).__name__ not in ["int", 'float']:
```

Comment:
It is better to add a judgement of whether decay is smaller than 0 or not since gradient of fm is easy to explode with the growth of learning_rate